src/creature.cpp

```diff
@@ -1017,19 +1019,38 @@ bool Creature::setFollowCreature(Creature* creature)
 			onWalkAborted();
 		}
 
-		hasFollowPath = false;
-		forceUpdateFollowPath = false;
 		followCreature = creature;
-		isUpdatingPath = true;
+		creature->addFollowedByCreature(this);
+		hasFollowPath = false;
+		g_dispatcher.addTask(createTask([id = getID()]() { g_game.updateCreatureWalk(id); }));
 	} else {
-		isUpdatingPath = false;
 		followCreature = nullptr;
 	}
 
 	onFollowCreature(creature);
 	return true;
 }
 
+// Pathfinding Functions
+void Creature::addFollowedByCreature(Creature* creature)
+{
+	if (creature) {
```

Comment:
I am not certain, but is it not possible to end up with a nullptr here? Maybe we can check` if (creature != nullptr)` or perhaps  `if (creature->isCreature())` , just to play it as safe as possible?

Reply:
I am pretty sure this is safe to remove the creature check all together. I will do so in the next commit.